src/NativeScript/inlineFunctions.js

```diff
@@ -14,6 +14,103 @@ Object.assign(global, {
     NSMakeRange(location, length) {
         return new NSRange({ location, length });
     },
+
+    // Decorators support    
+    __decorate(decorators, target, key, desc) {
+        var c = arguments.length;
+        var r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc;
+        var d;
+
+        if (typeof global.Reflect === "object" && typeof global.Reflect.decorate === "function") {
+            r = global.Reflect.decorate(decorators, target, key, desc);
+        }
+        else {
+            for (var i = decorators.length - 1; i >= 0; i--) {
+                if (d = decorators[i]) {
+                    r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
+                }
+            }
+        }
+        return c > 3 && r && Object.defineProperty(target, key, r), r;
+    },
+    __param(paramIndex, decorator) {
+        return function (target, key) {
+            decorator(target, key, paramIndex);
+        }
+    },  
+
+    // Decorators  
+    ObjC(name) {
+        var additionalParams = [];
+        var hasName = (name !== undefined && typeof name === "string");
+        var argsShift = hasName ? 1 : 0;
+        for (var i = argsShift; i < arguments.length; i++) {
+            additionalParams[i - argsShift] = arguments[i];
+        }
+
+        return function (target, propertyKey, descriptor) {
+            if (propertyKey === undefined) { // Class decorator
+                var implementation = {};
+                var baseTarget = Object.getPrototypeOf(target);
+                
+                for (var prop in target.prototype) {
+                    if (target.prototype.hasOwnProperty(prop)) {
+                        implementation[prop] = target.prototype[prop];
+                    }
+                }
+
+                var newName = hasName ? name : target.name;                
+                var extended = baseTarget.extend(implementation, {
```

Comment:
I see the benefit of specifying the name in the decorator and I am not against it but I don't agree with the current implementation. I suggest refactoring the [`ObjCClassBuilder`](https://github.com/NativeScript/ios-runtime/blob/master/src/NativeScript/ObjC/Inheritance/ObjCClassBuilder.h) to not require the name of the native class so early but this may be a bit tricky and is a good candidate for another PR.